packages/edit-site/src/components/url-query-controller/index.js

```diff
@@ -47,28 +47,19 @@ export default function URLQueryController() {
 
 function useCurrentPageContext() {
 	return useSelect( ( select ) => {
-		const {
-			getTemplateId,
-			getTemplatePartId,
-			getTemplateType,
-			getPage,
-		} = select( 'core/edit-site' );
+		const { getEditedPostType, getEditedPostId, getPage } = select(
+			'core/edit-site'
+		);
 
 		const page = getPage();
-		const templateType = getTemplateType();
-		const templateId = getTemplateId();
-		const templatePartId = getTemplatePartId();
-
-		let _postId, _postType;
+		let _postId = getEditedPostId(),
+			_postType = getEditedPostType();
+		// This doesn't seem right to me,
+		// we shouldn't be using the "page" and the "template" in the same way.
+		// This need to be investigated.
```

Comment:
I just worked a bit in this code here: https://github.com/WordPress/gutenberg/pull/27844, and this indeed needs deeper investigation. It touches the whole FSE navigation and as is now is buggy in the prioritization of `pageContext` due to the same handling (page/post - template/template-part).